src/components/PopupWithForm.js

```diff
@@ -1,20 +1,27 @@
-import Popup from './Popup';
+import Popup from './Popup.js';
 
 export default class PopupWithForm extends Popup {
     constructor(popupSelector, handleFormSubmit) {
         super({ popupSelector });
         this._popupForm = this._popupElement.querySelector(".modal__form");
         this._handleFormSubmit = handleFormSubmit;
         this._inputEls = [...this._popupForm.querySelectorAll(".modal__input")];
+        this._submitButton = this._popupForm.querySelector(".modal__button");
+        this._submitButtonText = this._submitButton.textContent;
     }
     
     setEventListeners() {
-        this._popupElement.addEventListener("submit", () => {
+        this._popupElement.addEventListener("submit", (e) => {
+            e.preventDefault();
             this._handleFormSubmit(this._getInputValues());
         });
         super.setEventListeners();
     }
 
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`